### ISSUE TITLE
Issue with latest VScode

### ISSUE THREAD

**Issue by DevDorrejo** [yamaton/vscode-h2o#4]
this extension delays the save.

```
2023-08-09 09:28:19.533 [error] TypeError: Cannot read properties of undefined (reading 'edit')
	at updateTree (/home/USER/.vscode/extensions/tetradresearch.vscode-h2o-0.2.9/out/extension.js:178:21)
	at r.edit [as value] (/home/USER/.vscode/extensions/tetradresearch.vscode-h2o-0.2.9/out/extension.js:184:13)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.x (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1972)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2188)
	at $.$acceptModelChanged (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:104:10065)
	at t.N (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:7983)
	at t.M (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:7749)
	at t.H (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:6830)
	at t.G (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:5906)
	at r.value (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:4736)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2119)
	at y.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:79:13931)
	at r.value (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:30355)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2119)
	at y.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:79:13931)
	at MessagePortMain.<anonymous> (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:28635)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)
```

**Comment by yamaton** [comment]
@DevDorrejo Thank you for using the extension, and reporting the issue!

I'll investigate. I have experienced delaying of the save multiple times, but never thought it's because of my extension...

**Comment by yamaton** [comment]
I updated with security patches + reduced log messages, and released it as 0.2.10. It's working fine for me without errors in the Runtime Status screen. (I did see the errors before the update, though.) Please let me know if you still have the issue. Thanks.

**Comment by DevDorrejo** [comment]
ok, thanks, i will monitor, if anything more happens, thanks for the hard work.

**Comment by DevDorrejo** [comment]
Hello,
i got delayed save.

when the delay saving finish, the OUTPUT > Extensions Host, start spamming the next (have to disable the extension):

```
2023-08-10 10:08:14.076 [error] TypeError: Cannot read properties of undefined (reading 'edit')
	at updateTree (/home/IND/wdorrejo/.vscode/extensions/tetradresearch.vscode-h2o-0.2.10/out/extension.js:178:21)
	at r.edit [as value] (/home/IND/wdorrejo/.vscode/extensions/tetradresearch.vscode-h2o-0.2.10/out/extension.js:184:13)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.x (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1972)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2188)
	at $.$acceptModelChanged (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:104:10065)
	at t.N (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:7983)
	at t.M (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:7749)
	at t.H (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:6830)
	at t.G (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:5906)
	at r.value (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:4736)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2119)
	at y.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:79:13931)
	at r.value (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:30355)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2119)
	at y.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:79:13931)
	at MessagePortMain.<anonymous> (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:28635)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)
2023-08-10 10:08:14.095 [error] TypeError: Cannot read properties of undefined (reading 'edit')
	at updateTree (/home/IND/wdorrejo/.vscode/extensions/tetradresearch.vscode-h2o-0.2.10/out/extension.js:178:21)
	at r.edit [as value] (/home/IND/wdorrejo/.vscode/extensions/tetradresearch.vscode-h2o-0.2.10/out/extension.js:184:13)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.x (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1972)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2188)
	at $.$acceptModelChanged (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:104:10065)
	at t.N (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:7983)
	at t.M (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:7749)
	at t.H (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:6830)
	at t.G (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:5906)
	at r.value (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:4736)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2119)
	at y.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:79:13931)
	at r.value (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:30355)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2119)
	at y.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:79:13931)
	at MessagePortMain.<anonymous> (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:28635)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)
2023-08-10 10:08:14.525 [error] TypeError: Cannot read properties of undefined (reading 'document')
	at /home/IND/wdorrejo/.vscode/extensions/xshrim.txt-syntax-0.2.2/src/highlight.js:39:71
	at Array.forEach (<anonymous>)
	at Highlight.updateDecorations (/home/IND/wdorrejo/.vscode/extensions/xshrim.txt-syntax-0.2.2/src/highlight.js:38:38)
	at Highlight.updateActive (/home/IND/wdorrejo/.vscode/extensions/xshrim.txt-syntax-0.2.2/src/highlight.js:86:10)
	at Timeout._onTimeout (/home/IND/wdorrejo/.vscode/extensions/xshrim.txt-syntax-0.2.2/src/extension.js:372:17)
	at listOnTimeout (node:internal/timers:559:17)
	at process.processTimers (node:internal/timers:502:7)
2023-08-10 10:08:14.654 [error] TypeError: Cannot read properties of undefined (reading 'edit')
	at updateTree (/home/IND/wdorrejo/.vscode/extensions/tetradresearch.vscode-h2o-0.2.10/out/extension.js:178:21)
	at r.edit [as value] (/home/IND/wdorrejo/.vscode/extensions/tetradresearch.vscode-h2o-0.2.10/out/extension.js:184:13)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.x (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1972)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2188)
	at $.$acceptModelChanged (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:104:10065)
	at t.N (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:7983)
	at t.M (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:7749)
	at t.H (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:6830)
	at t.G (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:5906)
	at r.value (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:4736)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2119)
	at y.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:79:13931)
	at r.value (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:30355)
	at f.w (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:1902)
	at f.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:63:2119)
	at y.fire (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:79:13931)
	at MessagePortMain.<anonymous> (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:28635)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)
```

**Comment by yamaton** [comment]
Appreciate the report. There must be an async bug, I suppose. I'll work on it.

**Comment by yamaton** [comment]
I guess the errors have been fixed in 0.2.11 released today. The bugs were long-standing ones, and I wanted to have noticed them earlier. Thank you @DevDorrejo !

**Comment by DevDorrejo** [comment]
You welcome, looks like it fixed, thank you very much.
